block/sync_service.go

```diff
@@ -130,18 +144,18 @@ func (bSyncService *BlockSyncService) WriteToBlockStoreAndBroadcast(ctx context.
 	return nil
 }
 
-func (bSyncService *BlockSyncService) isInitialized() bool {
-	return bSyncService.blockStore.Height() > 0
+func (syncService *SyncService[H]) isInitialized() bool {
+	return syncService.store.Height() > 0
 }
 
 // Start is a part of Service interface.
-func (bSyncService *BlockSyncService) Start() error {
+func (syncService *SyncService[H]) Start() error {
 	// have to do the initializations here to utilize the p2p node which is created on start
-	ps := bSyncService.p2p.PubSub()
-	chainIDBlock := bSyncService.genesis.ChainID + "-block"
+	ps := syncService.p2p.PubSub()
+	chainIDBlock := syncService.genesis.ChainID + "-" + syncService.prefix
```

Comment:
nit: modify `chainIDBlock` to `newChainID`
```suggestion
	newChainID := syncService.genesis.ChainID + "-" + syncService.prefix
```